corehq/apps/users/static/users/js/roles.js

```diff
@@ -192,9 +193,37 @@ hqDefine('users/js/roles',[
             },
         };
 
+        self.allowSubmit = ko.computed(function () {
+            if (!self.roleBeingEdited()) {
+                return false;
+            }
+            if (!self.roleBeingEdited().name()) {
+                return false;
+            }
+            return true;
+        });
+
+        self.submitNewRole = function () {
+            // moved saveOptions inline
+            $.ajax({
+                method: 'POST',
+                url: o.saveUrl,
+                data: JSON.stringify(UserRole.unwrap(self.roleBeingEdited)),
+                dataType: 'json',
+                success: function (data) {
+                    self.addOrReplaceRole(data);
+                    self.unsetRoleBeingEdited();
```

Comment:
Out of curiosity, is this what closes the modal?

Reply:
To be honest I'm not entirely sure what's closing the modal, although that seems like where it's coming from.